aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -67,11 +69,19 @@ module aptos_framework::object {
     /// A shared resource group for storing object resources together in storage.
     struct ObjectGroup { }
 
-    /// Type safe way of designate an object as at this address.
+    /// Type-safe way of designate an object as at this address.
     struct ObjectId has copy, drop, store {
         inner: address,
     }
 
+    /// A typed ObjectId -- these can only provide guarantees based upon the underlying data type,
+    /// that is the validity of T existing at an address is something that cannot be verified by
+    /// any other module than the module that defined T. Similarly, the module that defines T can
```

Comment:
Based on what you are saying here, you can just drop the exist_at which is anyway (I think) not cleanly implemented, as the current impl in the other PR appears to shortcut layering levels.

I think it is probably OK to leave this references weak for now. Guarantees when those objects exist might be established in other ways, like resource group/object types. Even without existence guarantee, type references are much better than untyped ones.